### PR TITLE
Store node ID's and weights separately in CompressedEdgeContainer

### PR DESCRIPTION
**Pull request by jakepruitt**
Removes CompressedEdges from the extractor and shared data format by
directly serializing vectors of node ID's, forward weights and reverse
weights for each node-based-edge

Refs #2575

## Tasklist

- [ ] ~~Refactor `GetUncompressed*` methods to return boost views to internal arrays~~ ticketing separately after @danpat tackles https://github.com/Project-OSRM/osrm-backend/issues/3072
- [x] Review from @TheMarex 

cc/ @danpat @daniel-j-h @MoKob

### CODE REVIEW

File: include/engine/datafacade/internal_datafacade.hpp [hunk 1]
```diff
@@ -727,20 +739,18 @@ class InternalDataFacade final : public BaseDataFacade
          * both forward and reverse segments along the same bi-
          * directional edge. The m_geometry_indices stores
          * refences to where to find the beginning of the bi-
-         * directional edge in the m_geometry_list vector.
+         * directional edge in the m_geometry_node_list vector.
          * */
-        const unsigned begin = m_geometry_indices.at(id);
-        const unsigned end = m_geometry_indices.at(id + 1);
+        const signed begin = m_geometry_indices.at(id);
+        const signed end = m_geometry_indices.at(id + 1);
```
Comment (TheMarex) [inline include/engine/datafacade/internal_datafacade.hpp]: Where is this change comming from? In general `signed` is quite unusual in C++, I would actually recommend doing `auto` everywhere in these functions.

File: include/engine/datafacade/internal_datafacade.hpp [hunk 2]
```diff
@@ -780,21 +786,19 @@ class InternalDataFacade final : public BaseDataFacade
...
          * refences to where to find the beginning of the bi-
-         * directional edge in the m_geometry_list vector. For
+         * directional edge in the m_geometry_rev_weight_list vector. For
          * reverse weights of bi-directional edges, edges 1 to
          * n-1 of that edge need to be read in reverse.
          */
-        const unsigned begin = m_geometry_indices.at(id);
-        const unsigned end = m_geometry_indices.at(id + 1) - 1;
+        const signed begin = m_geometry_indices.at(id);
+        const signed end = m_geometry_indices.at(id + 1) - 1;
```
Comment (TheMarex) [inline include/engine/datafacade/internal_datafacade.hpp]: Same thing with signed.

File: include/engine/datafacade/shared_datafacade.hpp [hunk 3]
```diff
@@ -592,21 +604,23 @@ class SharedDataFacade final : public BaseDataFacade
...
+                  result_weights.begin());
 
-        std::for_each(m_geometry_list.rbegin() + (m_geometry_list.size() - end),
-                      m_geometry_list.rbegin() + (m_geometry_list.size() - begin),
-                      [&](const osrm::extractor::CompressedEdgeContainer::CompressedEdge &edge) {
-                          result_weights.emplace_back(edge.reverse_weight);
-                      });
+        std::cout << "Should be: " << end - begin << std::endl;
+        std::cout << "Result weights length: " << result_weights.size() << std::endl;
+        std::cout << "Beginning weights length: " << m_geometry_rev_weight_list.size() << std::endl;
```
Comment (TheMarex) [inline include/engine/datafacade/shared_datafacade.hpp]: Left-over debugging code?

Comment (jakepruitt) [inline include/engine/datafacade/shared_datafacade.hpp]: 😧 yep

**Pull request merged**